streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -98,4 +147,41 @@ public String toString() {
         }
         return sb.toString();
     }
+
+    protected class NodeMetrics  {
+        final StreamsMetrics metrics;
+        final String metricGrpName;
+        final Map<String, String> metricTags;
+
+        final Sensor nodeProcessTimeSensor;
+        final Sensor nodePunctuateTimeSensor;
+        final Sensor nodeThroughputSensor;
+        final Sensor nodeCreationSensor;
+        final Sensor nodeDestructionSensor;
+
+
+        public NodeMetrics(StreamsMetrics metrics, String name, String sensorNamePrefix) {
+            String scope = "processor-node";
+            this.metrics = metrics;
+            this.metricGrpName = "stream-processor-node-metrics";
+            this.metricTags = new LinkedHashMap<>();
+            this.metricTags.put("processor-node-id", name);
+
+            // these are all latency metrics
+            this.nodeProcessTimeSensor = metrics.addLatencySensor(scope, sensorNamePrefix + name, "process", Sensor.RecordLevel.SENSOR_DEBUG, "processor-node-id", name);
```

Comment:
I left a follow-up comment about not needing to include `name` in the prefix anymore, let me know what do you think.